lib/Doctrine/DBAL/Connection.php

```diff
@@ -1597,7 +1593,7 @@ public function ping()
         }
 
         try {
-            $this->query($this->platform->getDummySelectSQL());
+            $this->query($this->getDatabasePlatform()->getDummySelectSQL());
```

Comment:
Why is this change necessary? `Connection::ping()` calls `Connection::connect()` as the beginning, so it ensures the platform is set.